lib/miq_automation_engine/engine/miq_ae_method_service/miq_ae_service_methods.rb

```diff
@@ -144,7 +144,7 @@ def self.create_provision_request(*args)
       # Need to add the username into the array of params
       # TODO: This code should pass a real username, similar to how the web-service
       #      passes the name of the user that logged into the web-service.
-      args.insert(1, User.lookup_by_userid("admin")) if args.kind_of?(Array)
+      args << User.lookup_by_userid("admin")
```

Comment:
This fixes two bugs:
1) args is always an array because it's splatted
2) it inserted the user as the second element in the array, so, if you had an empty array, it would return `[nil, user]` on this line.